src/dotnet-new3/ParserFactory.cs

```diff
@@ -20,22 +22,34 @@ internal static Parser CreateParser(Command command, bool disableHelp = false)
             //.UseParseDirective()
             //.UseSuggestDirective()
             .UseParseErrorReporting()//TODO: discuss with SDK if it is possible to use it.
-            //TODO: implement when help is done
-            //.UseHelpBuilder((context) =>
-            //{
-            //    var hb = new HelpBuilder(LocalizationResources.Instance);
-            //    hb.Customize(command, (result) => )
-            //    return hb;
-            // })
             .DisablePosixBundling();
 
             if (!disableHelp)
             {
-                builder = builder.UseHelp();
+                builder = builder.UseHelp(CustomHelpLayout());
             }
             return builder.Build();
         }
 
+        private static IEnumerable<HelpDelegate> CustomHelpLayout()
+        {
+            yield return (HelpContext context) =>
+            {
+                if (context.ParseResult.CommandResult.Command is ICustomHelp custom)
+                {
+                    custom.WriteHelp(context);
+                }
+                else
+                {
+                    foreach (var section in HelpBuilder.DefaultLayout())
+                    {
+                        section.Invoke(context);
+                        context.Output.WriteLine();
+                    }
+                }
+            };
+        }
+
```

Comment:
Maybe `UseHelp` should accept a `Func<HelpContext, IEnumerable<HelpDelegate>>` rather than  `IEnumerable<HelpDelegate>` ? This way you could hoist the decision up and do something like this: 

```cs
if (context.ParseResult.CommandResult.Command is ICustomHelp custom)
{
    return CustomLayout();
}
else
{
    return context.HelpBuilder.DefaultLayout();
}
```